src/components/select/select.jsx

```diff
@@ -117,6 +117,8 @@ const Select = forwardRef(function Select(props, ref) {
   useEffect(() => {
     if (!mounted.current) return;
 
+    if (props.value === undefined) return;
```

Comment:
As above. Without this, our default state becomes undefined, which seems like a breaking of contract, but also means that if `undefined` is intentionally passed for `props.value`, `onChange` will not fire.